src/java/com/rapleaf/jack/BaseDatabaseConnection.java

```diff
@@ -165,4 +181,21 @@ public void close() throws IOException {
       throw new IOException(e);
     }
   }
+
+  private boolean disabled() {
```

Comment:
not to hate on Optional, but I feel like boolean might be more straighforward here

````
boolean disabledBySystemProperty = false;
      try {
        disabledBySystemProperty = Boolean.parseBoolean(System.getProperty(DISABLE_PROPERTY));
       } catch (IllegalArgumentException e) {
         LOG.error("blah blah ignoring", e);
       }

      return disabledBySystemProperty || disabledByEnvVar;
````